### DISCUSSION TITLE
[FEATURE] State Estimation Features

### DISCUSSION THREAD

**Discussion by Welthulk** [PowerGridModel/.github#21] [Q&A] [answered]
I have read the manuals, but unfortunately I cannot find out which features are supported by the State Estimation. 
Questions:
Are subnets supported?
Are island networks supported?
How are couplings on busbars modeled?
Are phase-shifting transformers supported?
Can transformer taps be estimated?
How are parallel lines and transformers handled?

Thank you very much for your answer

**Comment by TonyXiang8787** [comment]
Hi @Welthulk,

I migrated your question into this discussion tickets. @nitbharambe can you help with the questions here?

**Comment by nitbharambe** [comment, marked as answer]
Thanks for asking these in depth questions, @Welthulk . 

1) The components within a subnet need to be selected manually by the user and islanded manually if required. Rest will be handled directly within the model.

2) Island networks are supported. If  there is atleast one source connected, there will be power flowing. Otherwise the components wont be `energized` and outputs are zeroed.

3) Couplings can be modelled by a `link` component, which is modelled as a very high admittance internally. Note: `link` component however does not support power sensors to be attached to measure power flowing from / to link. (Also, if there is a detailed model of coupling's impedance, then a line can also be used by specifying R, L , C in `pi` form. Same applies if there is a need for modelling an `impedance` between two buses/nodes.)

4) We don't support phase shifter transformers yet: https://github.com/PowerGridModel/power-grid-model/issues/188

5) We don't support tap position as a measured quantity in state estimation either: https://github.com/PowerGridModel/power-grid-model/issues/132

6) Parallel Lines and transformers are supported. You just have to create a copy of the component.

Let us know if you wish to know more details!

**Comment by Welthulk** [comment]
Thank you very much for the very prompt response. There may have been a translation error regarding the topic of couplings between busbars. I meant to ask if there is a separate model for the circuit breakers between busbars. Most state estimations model the  connection between busbars as a very short line with very low ohmic losses (approximately 0.1 Ohm). At the moment, my questions are answered. Thank you.

**Comment by TonyXiang8787** [comment]
Hi @Welthulk, nice to see your questions are answered. We usually keep the discussion (question) open so other people can read it. The question has been already marked as answered.
